service/status-report/main.go

```diff
@@ -215,6 +221,29 @@ func setClusterInformation(ctx context.Context, cl client.Client, status interfa
 
 }
 
+func setStorageQuotaUtilizationRatio(ctx context.Context, cl client.Client, status interfaces.StorageClientStatus) {
+	clusterResourceQuota := &quotav1.ClusterResourceQuota{}
+	clusterResourceQuota.Name = status.GetClientName()
+
+	// No need to check for NotFound because unlimited quota client will not have CRQ resource
+	if err := cl.Get(ctx, client.ObjectKeyFromObject(clusterResourceQuota), clusterResourceQuota); client.IgnoreNotFound(err) != nil {
+		klog.Warningf("Failed to get clusterResourceQuota %q: %v", clusterResourceQuota.Name, err)
+	}
+
+	if clusterResourceQuota.Status.Total.Hard != nil {
```

Comment:
Are we sure none of parts of `.Status.Total.Hard` is a pointer? If any is you will need a `nil` pointer check

Reply:
Yes, I have verified with multiple scenario.

Reply:
There is no need to look at scenarios, just jump the type definition of `Status` and see if `Status` or `Total` are defined as pointer/interfaces or not

Reply:
Done.